clusters/resource_cluster.go

```diff
@@ -241,14 +241,54 @@ func resourceClusterUpdate(ctx context.Context, d *schema.ResourceData, c *commo
 	common.DataToStructPointer(d, clusterSchema, &cluster)
 	var clusterInfo ClusterInfo
 	var err error
+
 	if hasClusterConfigChanged(d) {
 		log.Printf("[DEBUG] Cluster state has changed!")
 		if err := cluster.Validate(); err != nil {
 			return err
 		}
 		cluster.ModifyRequestOnInstancePool()
 		fixInstancePoolChangeIfAny(d, &cluster)
-		clusterInfo, err = clusters.Edit(cluster)
+
+		numberOfClusterConfigChanges := 0
```

Comment:
why do we need `numberOfClusterConfigChanges`? it looks redundant. `k == "library" || k == "is_pinned"` are checked by `hasClusterConfigChanged` already.

Reply:
I feel like this is also a bit confusing, do you have a solution to simplify the boolean calculation here ? I am partial towards reverting back to computing the number of config fields changed